tfhe/src/integer/server_key/radix_parallel/add.rs

```diff
@@ -338,18 +339,47 @@ impl ServerKey {
     ///
     /// Requires the blocks to have at least 4 bits
     pub(crate) fn compute_carry_propagation_parallelized_low_latency(
+        &self,
+        generates_or_propagates: Vec<Ciphertext>,
+    ) -> (Vec<Ciphertext>, Ciphertext) {
```

Comment:
At the integer level, having a Ciphertext seems ok to me as this is likely to be used in cmux

If it were an integer, would it be 1 block sized, or same num blocks a input like I did for comparisons ?

Talking of comparisons, they return an integer, but in practice a lot of time they are used you do cmp_result.blocks[0], because you know others are trivials.


A boolean like integer will likely be needed at some point, probably when we can migrate the hlapi to being integers only

Reply:
I believe the boolean integer would be a single shortint Ciphertext underneath, just that we have two ways of representing booleans with this code change/comparisons